hazelcast/include/hazelcast/client/internal/socket/BaseSocket.h

```diff
@@ -263,12 +265,12 @@ class BaseSocket : public hazelcast::client::socket
                   connection->last_write_time(std::chrono::steady_clock::now());
 
                   if (!this->outbox_.empty()) {
-                      do_write(connection, invocation);
+                      do_write(connection, outbox_[0].invocation);
```

Comment:
I think it is no longer needed to pass `invocation` to the `do_write` method since its lifetime is already extended by appending `outbox_` queue. So I think it is better to remove `invocation` parameter from `do_write` function.

Reply:
fixed